src/java.base/share/classes/java/lang/ThreadGroup.java

```diff
@@ -658,10 +658,16 @@ private List<ThreadGroup> synchronizedSubgroups() {
 
     /**
      * Returns a snapshot of the subgroups as an array, used by JVMTI.
+     * WARNING: Make sure this method does not trigger any class loading.
```

Comment:
Can you say why briefly here?  Because it's called during thread startup and may deadlock with ClassLoad events.  I see the comment in the test but not sure if someone will find it, except through git blame, which might be enough.

Reply:
I was trying to avoid getting too much into the weeds with the explanation. There's always that one extra bit of information that would be useful, but after adding that then there's another, and you need to decide when to stop. How about the following (and I really don't want to go any further than this):

     * WARNING: Make sure this method does not trigger any class loading,
     * because a ClassPrepareEvent can deadlock the debugger and debug agent.